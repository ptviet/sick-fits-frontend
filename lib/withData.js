import withApollo from 'next-with-apollo';
import ApolloClient from 'apollo-boost';
import { endpointProd } from '../config';
import { LOCAL_STATE_QUERY } from '../components/Cart';

function createClient({ headers }) {
  return new ApolloClient({
    // uri: process.env.NODE_ENV === 'development' ? endpoint : endpointProd,
    uri: 'https://sp-sick-fits-yoga-prod.herokuapp.com',
    request: operation => {
      operation.setContext({
        fetchOptions: {
          credentials: 'include'
        },
        headers
      });
    },
    // Local data
    clientState: {
      resolvers: {
        Mutation: {
          toggleCart(_, variables, { cache }) {
            // Read the cartOpen value from the cache
            const { cartOpen } = cache.readQuery({ query: LOCAL_STATE_QUERY });
            // Write new cartOpen state
            const data = {
              data: {
                cartOpen: !cartOpen
              }
            };
            cache.writeData(data);
            return data;
          }
        }
      },
      defaults: {
        cartOpen: false
      }
    }
  });
}

export default withApollo(createClient);
