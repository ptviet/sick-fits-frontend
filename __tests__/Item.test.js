import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Item from '../components/Item';
import { fakeItem } from '../lib/testUtils';

describe('<Item />', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Item item={fakeItem()} />);
  });

  it('should render', () => {
    expect(wrapper.exists()).toBeTruthy();
  });

  it('should match the snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should display properly', () => {
    expect(wrapper.find('Title a').text()).toBe(fakeItem().title);

    expect(wrapper.find('img').props().src).toBe(fakeItem().image);
    expect(wrapper.find('img').props().alt).toBe(fakeItem().title);

    const PriceTag = wrapper.find('PriceTag');
    expect(PriceTag.children().text()).toBe('$50');

    expect(wrapper.find('p').text()).toBe(fakeItem().description);

    expect(wrapper.find('.buttonList').children()).toHaveLength(3);
    expect(wrapper.find('.buttonList').find('Link')).toHaveLength(1);
    expect(
      wrapper
        .find('.buttonList')
        .find('Link a')
        .text()
    ).toBe('Edit ✏️');

    expect(wrapper.find('AddToCart')).toHaveLength(1);
    expect(wrapper.find('DeleteItem')).toHaveLength(1);
  });
});
