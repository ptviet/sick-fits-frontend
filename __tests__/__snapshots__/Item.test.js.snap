// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item /> should match the snapshot 1`] = `
<ItemStyles__Item>
  <img
    alt="dogs are best"
    src="dog-small.jpg"
  />
  <Title>
    <Link
      href={
        Object {
          "pathname": "/item",
          "query": Object {
            "id": "abc123",
          },
        }
      }
    >
      <a>
        dogs are best
      </a>
    </Link>
  </Title>
  <PriceTag>
    $50
  </PriceTag>
  <p>
    dogs
  </p>
  <div
    className="buttonList"
  >
    <Link
      href={
        Object {
          "pathname": "/update",
          "query": Object {
            "id": "abc123",
          },
        }
      }
    >
      <a>
        Edit ✏️
      </a>
    </Link>
    <AddToCart
      id="abc123"
    />
    <DeleteItem
      id="abc123"
    >
      Delete ❌
    </DeleteItem>
  </div>
</ItemStyles__Item>
`;
